check/setup_test.go

```diff
@@ -273,6 +273,11 @@ func Test_deleteTopic_WhenCreateDeleteNodeFails_ReturnsError(t *testing.T) {
 
 	connection.EXPECT().DeleteTopic("health-check", gomock.Any()).Return(errors.New("New error"))
 
+	// here we are also testing that retriers do 3 tries before return an error
+	connection.EXPECT().DeleteTopic("health-check", gomock.Any()).Return(errors.New("New error"))
```

Comment:
I think we should parameterise the method (and maybe the healthcheck itself), so that testing can be simplified and doesn't depend on the waiting strategy